staffs/views.py

```
@@ -1,3 +1,380 @@
+)
+
+class TeacherViewSet(StatusManagerMixin, viewsets.ModelViewSet):
+    """
+    ViewSet for managing Teacher profiles and related academic metrics.
+    
+    Provides standard CRUD operations along with specialized actions for
+    retrieving course assignments, teaching materials, and workload analytics.
+    """
+    queryset = Teacher.objects.all().select_related('user').order_by('-id')
+    related_lookups = ['user']
+    allow_status_on = ['list', 'materials', 'courses', 'assignments']
```

Ruff (0.15.12)

[warning] Mutable default value for class attribute

(RUF012)